packages/server/src/adapters/lambda/index.ts

```diff
@@ -0,0 +1,156 @@
+import type {
+  Context as APIGWContext,
+  APIGatewayProxyEvent,
+  APIGatewayProxyEventV2,
+  APIGatewayProxyResult,
+  APIGatewayProxyStructuredResultV2,
+} from 'aws-lambda';
+import { TRPCError, resolveHTTPResponse } from '../..';
+import { HTTPHeaders, HTTPRequest } from '../../http/internals/types';
+import type { HTTPResponse } from '../../http/internals/types';
+import { AnyRouter, inferRouterContext } from '../../router';
+import {
+  APIGatewayEvent,
+  APIGatewayResult,
+  AWSLambdaOptions,
+  UNKNOWN_PAYLOAD_FORMAT_VERSION_ERROR_MESSAGE,
+  isPayloadV1,
+  isPayloadV2,
+} from './utils';
+
+export type { CreateLambdaContextOptions, AWSLambdaOptions } from './utils';
+
+function lambdaEventToHTTPRequest(event: APIGatewayEvent): HTTPRequest {
+  const query = new URLSearchParams();
+  for (const [key, value] of Object.entries(
+    event.queryStringParameters ?? {},
+  )) {
+    if (typeof value !== 'undefined') {
+      query.append(key, value);
+    }
+  }
+
+  return {
+    method: getHTTPMethod(event),
+    query: query,
+    headers: event.headers,
+    body: event.body,
+  };
+}
+
+function getHTTPMethod(event: APIGatewayEvent) {
+  if (isPayloadV1(event)) {
+    return event.httpMethod;
+  } else if (isPayloadV2(event)) {
+    return event.requestContext.http.method;
+  } else {
+    throw new TRPCError({
+      code: 'INTERNAL_SERVER_ERROR',
+      message: UNKNOWN_PAYLOAD_FORMAT_VERSION_ERROR_MESSAGE,
+    });
+  }
```

Comment:
Some unnecessary if/else

```suggestion
  if (isPayloadV1(event)) {
    return event.httpMethod;
  }
  if (isPayloadV2(event)) {
    return event.requestContext.http.method;
  }
  throw new TRPCError({
    code: 'INTERNAL_SERVER_ERROR',
    message: UNKNOWN_PAYLOAD_FORMAT_VERSION_ERROR_MESSAGE,
  });
```